lib/modules/datasource/flutter-version/index.ts

```diff
@@ -0,0 +1,58 @@
+import { Datasource } from '../datasource';
+import type { GetReleasesConfig, ReleaseResult } from '../types';
+import type { FlutterResponse } from './types';
+
+export class FlutterDatasource extends Datasource {
+  static readonly id = 'flutter-version';
+
+  constructor() {
+    super(FlutterDatasource.id);
+  }
+
+  override readonly customRegistrySupport = false;
+
+  override readonly defaultRegistryUrls = ['https://storage.googleapis.com'];
+
+  override readonly caching = true;
+
+  async getReleases({
+    registryUrl,
+  }: GetReleasesConfig): Promise<ReleaseResult | null> {
+    // istanbul ignore if
+    if (!registryUrl) {
+      return null;
+    }
+    const result: ReleaseResult = {
+      homepage: 'https://flutter.dev',
+      sourceUrl: 'https://github.com/flutter/flutter',
+      registryUrl,
+      releases: [],
+    };
+    try {
+      const resp = (
+        await this.http.getJson<FlutterResponse>(
+          `${registryUrl}/flutter_infra_release/releases/releases_linux.json`
+        )
+      ).body;
+      const stableRegex = /^\d+\.\d+\.\d+$/;
+      result.releases = resp.releases
+        // The API response contains a stable version being released as a non-stable
+        // release. And so we filter out these releases here.
+        .filter(({ version, channel }) => {
```

Comment:
After looking into the API response from https://storage.googleapis.com/flutter_infra_release/releases/releases_linux.json in more detail, it looks like there are stable versions being released as a non-stable release. See version `2.5.0` for example, it has two releases in the response, one with the `stable` channel and one with `beta`.

So I added this filter here to only retrieve the stable release for stable versions, i.e. versions with `<major>.<minor>.<patch>`.